README.md

```diff
@@ -18,3 +18,397 @@ We are looking for more hands to join in - or review the task list so that we ca
+    )
+    parser.add_argument(
+        "--node",
+        "-w",
+        type=str,
+        help="The specified nodes to use. Same format as the "
+        "return of 'sinfo'. Default: ''.",
+    )
+    parser.add_argument(
+        "--num-gpus",
+        type=int,
+        default=0,
+        help="Number of GPUs to use in each node. (Default: 0)",
+    )
+    parser.add_argument(
+        "--partition",
+        "-p",
+        type=str,
+    )
+    parser.add_argument(
+        "--activation-script",
+        type=str,
+        help="The full path to the shell script to source for activating the environment (e.g., /data/activate.sh).",
+        default="",
+    )
+    parser.add_argument(
+        "--command",
+        type=str,
+        required=True,
+        help="The command you wish to execute. For example: "
+        " --command 'python test.py'. "
+        "Note that the command must be a string.",
+    )
+    parser.add_argument(
+        "--dry-run",
+        action="store_true",
+        help="If set, generate the submission script but do not submit it."
+    )
+
+    args = parser.parse_args()
+
+    if args.node:
+        node_info = "#SBATCH -w {}".format(args.node)
+    else:
+        node_info = ""
+
+    job_name = "{}_{}".format(
+        args.exp_name, time.strftime("%m%d-%H%M", time.localtime())
+    )
+    
+    partition_option = (
+        "#SBATCH --partition={}".format(args.partition) if args.partition else ""
+    )
+
+    if args.activation_script:
+        load_env_command = f"source {args.activation_script}"
+    else:
+        load_env_command = "# No activation script provided."
+
+    if not template_file.is_file():
+        print(f"Error: Template file not found at {template_file}")
+        sys.exit(1)
+        
+    with open(template_file, "r") as f:
+        text = f.read()
+
+    text = text.replace(JOB_NAME, job_name)
+    text = text.replace(NUM_NODES, str(args.num_nodes))
+    text = text.replace(NUM_GPUS_PER_NODE, str(args.num_gpus))
+    text = text.replace(PARTITION_OPTION, partition_option)
+    text = text.replace(COMMAND_PLACEHOLDER, str(args.command))
+    text = text.replace(LOAD_ENV, load_env_command)
+    text = text.replace(GIVEN_NODE, node_info)
+    text = text.replace(
+        "# THIS FILE IS A TEMPLATE AND IT SHOULD NOT BE DEPLOYED TO PRODUCTION!",
+        "# THIS FILE IS MODIFIED AUTOMATICALLY FROM TEMPLATE AND SHOULD BE "
+        "RUNNABLE!",
+    )
+
+    script_file = "{}.sh".format(job_name)
+    with open(script_file, "w") as f:
+        f.write(text)
+
+    if args.dry_run:
+        print("--- DRY RUN MODE ---")
+        print(f"Job script '{script_file}' has been generated but NOT submitted.")
+        print("You can inspect the script and submit it manually with:")
+        print(f"  sbatch {script_file}")
+    else:
+        print("Starting to submit job!")
+        result = subprocess.run(
+            ["sbatch", script_file], 
```

Comment:
It'd be helpful to have the `sbatch` installation described above as well

Reply:
it is in the docker container

Reply:
In the real prod environment the user will not install the HPC/supercomputer just use it.

Reply:
Okay I see so this is meant to be run inside the docker container